src/box/iproto.cc

```diff
@@ -2095,9 +2095,16 @@ tx_process_misc(struct cmsg *m)
 					   ::schema_version);
```

Comment:
`The built-in net.box connector sets this feature bit.` - this can't be done either. You are going to break the existing code which might return error objects and expect strings on the client side. Not the netbox itself, but the code which uses it, and returns `nil, err` from some calls existing `type(err) == 'string'`.

It wasn't done before not because we couldn't figure out how to do that. But because on the contrary with stuff like UUIDs and decimals you couldn't break anything unless you start using them. The error type on the server was added super long time, and by suddenly changing how it is encoded by default some of existing users can actually be broken.

If you want to force push that, you need to talk to Mons and Dmitry. And probably some of solutions: Yaroslav, Oleg. As main users of this all.

Reply:
I think we should enable the feature by default right now and disable it only if there are complaints. I think so, because:
 - An error object is convertible to a string so unless someone explicitly asserts that its type is 'string', there will be no problems.
 - Error objects (not strings) are returned by all Tarantool functions. The fact that they are converted to strings by net.box is a bug that can finally be fixed. Users have to implement their own error modules because of this bug.
 - There's a one-line workaround to disable the feature: `msgpack.cfg{encode_error_as_ext = false}`. So, should anything break, the user can always revert to the old behavior.
 - Connectors that don't support the `MP_ERROR` MsgPack extension will continue working as before, because for them the feature will be disabled (since they don't set the `IPROTO_FEATURE_ERROR_EXTENSION` bit in the `IPROTO_ID` request).

Discussed with
- @unera: [ok](https://user-images.githubusercontent.com/1114511/133604397-3f82534d-9303-40bf-8400-226ed3585a43.png)
- @olegrok: [ok](https://user-images.githubusercontent.com/1114511/133605316-1cfe4c77-9c99-4589-a674-41eff0171070.png)
- @rosik: ok, but it turned out that the `IPROTO_ID` command breaks [cartridge-remote-control](https://github.com/tarantool/cartridge/blob/master/cartridge/remote-control.lua#L288-L291), which mocks Tarantool server. Yaroslav will fix the mocker to support `IPROTO_ID` and then re-run cartridge tests on this branch. I'll follow up on this here.

Reply:
@rosik will fix cartridge-remote-control by mocking the `IPROTO_ID` command. It's okay to merge this PR before the fix, because cartridge isn't tested against Tarantool master.

Note, this PR shouldn't break the case when new net.box connects to cartridge running Tarantool < 2.10.0, because new net.box sends `IPROTO_ID` only if the server version received in a greeting message is >= 2.10.0.